simpleocr/files.py

```diff
@@ -45,45 +32,152 @@ def write(self):
+        """
+        path = Image.get_absolute_path(path)
+        array = cv2.imread(path)
+        return Image(array=array, path=path)
+
+    @staticmethod
+    def from_pil(pillow):
+        """ Create an Image object based off of a Pillow Image """
+        return Image(Image.get_array_from_pil(pillow))
+
+    @staticmethod
+    def get_absolute_path(image):
+        """
+        Get the absolute path for an image. Valid inputs are:
+        - Relative path with or without extension
+        - File in DATA_DIRECTORY with or without extension
+        - Absolute path with or without extension
+        :param image: image path in str type
+        :return: The absolute image path
+        """
+        # If the path exists, return the path, but make sure it's an absolute path first
+        if os.path.exists(image):
+            return os.path.abspath(image)
+        # Try to find the file with the passed path with the various extensions
+        image_with_extension = try_extensions(IMAGE_EXTENSIONS, os.path.splitext(image)[0])
+        if image_with_extension:
+            return os.path.abspath(image_with_extension)
+        # The file must be in the data directory if it has not yet been found
+        image_basename = os.path.basename(image)
+        image_datadir = try_extensions(IMAGE_EXTENSIONS, os.path.join(DATA_DIRECTORY, image_basename))
+        if image_datadir:
+            return os.path.abspath(image_datadir)
+        # The file cannot be found, so raise a FileNotFound Error
+        raise Image.FileNotFound
+
+    @staticmethod
+    def get_array_from_pil(pillow):
+        """
+        Returns an OpenCV compatible array from a Pillow Image object
+        :param pillow: pillow image object
+        :return: OpenCV compatible numpy array
+        """
+        imagefile = numpy.array(pillow)
+        return imagefile[:, :, ::-1].copy()
 
     @property
     def is_grounded(self):
-        """checks if this file is grounded"""
+        """
+        :return: True if Image is grounded
+        """
         return not (self.ground is None)
 
-    def set_ground(self, segments, classes, write_file=False):
-        """creates the ground, saves it to a file"""
-        if self.is_grounded:
+    @property
+    def has_file_reference(self):
+        """
+        :return: True if Image has a file reference
+        """
+        return self.path is not None
+
+    def set_ground(self, segments, classes, write=False):
+        """
+        Creates the ground data in memory, optionally writing it to a ground file,
+        if write is set to True
+        :param segments: segments to be grounded
+        :param classes: classes to be grounded
+        :param write: if True, writes the ground data to a ground file
+        :return: None
+        """
+        if self.is_grounded and self._debug:
             print("Warning: grounding already grounded file")
         self.ground = GroundFile(self.ground_path)
```

Comment:
We'll need some mechanism to handle this. We shouldn't create a GroundFile if the Image doesn't have a file.
We probably need `Ground` and `GroundFile` as well